vendor/k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go

```diff
@@ -392,6 +392,10 @@ func (u *Unstructured) GetCreationTimestamp() metav1.Time {
 
 func (u *Unstructured) SetCreationTimestamp(timestamp metav1.Time) {
 	ts, _ := timestamp.MarshalQueryParameter()
+	if len(ts) == 0 {
+		u.setNestedField(nil, "metadata", "creationTimestamp")
```

Comment:
@liggitt this fixes the issue I had mentioned in our conversation earlier.
Exporting unstructured objects with an empty string as the value of their
`creationTimestamp` field was causing an error [1] when trying to re-import
those objects with `oc create`.

```
$ oc export bc/foo > foo.yaml
...
$ oc create -f foo.yaml
Error from server (BadRequest): BuildConfig in version "v1" cannot be handled as a BuildConfig: parsing time "" as "2006-01-02T15:04:05Z07:00": cannot parse "" as "2006"
```

Reply:
@ironcladlou you spent a lot of time in here.  This look good?

Reply:
Assigning a nil key to the internal map seems to be a perpetuation of the problem outlined in https://github.com/kubernetes/kubernetes/issues/48211 (related to several bugs).

Potentially diverging from upstream by carrying a patch for this particular change seems riskier than usual given the history of nuanced serialization problems in this type. My suggestion would be to make an upstream commit a prerequisite.

Reply:
> Assigning a nil key to the internal map seems to be a perpetuation of the problem outlined in kubernetes/kubernetes#48211 (related to several bugs).
> 
> Potentially diverging from upstream by carrying a patch for this particular change seems riskier than usual given the history of nuanced serialization problems in this type. My suggestion would be to make an upstream commit a prerequisite.

@ironcladlou you'll be the likely reviewer upstream.  Will it merge upstream?

Reply:
@juanvallejo, can you extract this patch into an upstream PR and include a unit test demonstrating that it fixes a bug?

Reply:
@ironcladlou 
Upstream PR opened here: https://github.com/kubernetes/kubernetes/pull/53464